src/binary/uint.rs

```diff
@@ -119,39 +127,39 @@ mod tests {
     #[test]
     fn test_read_one_byte_uint() {
         let data = &[0b1000_0000];
-        let uint = UInt::read(&mut Cursor::new(data), data.len()).expect(READ_ERROR_MESSAGE);
+        let uint = DecodedUInt::read(&mut Cursor::new(data), data.len()).expect(READ_ERROR_MESSAGE);
         assert_eq!(uint.size_in_bytes(), 1);
         assert_eq!(uint.value(), 128);
     }
 
     #[test]
     fn test_read_two_byte_uint() {
         let data = &[0b0111_1111, 0b1111_1111];
-        let uint = UInt::read(&mut Cursor::new(data), data.len()).expect(READ_ERROR_MESSAGE);
+        let uint = DecodedUInt::read(&mut Cursor::new(data), data.len()).expect(READ_ERROR_MESSAGE);
         assert_eq!(uint.size_in_bytes(), 2);
         assert_eq!(uint.value(), 32_767);
     }
 
     #[test]
     fn test_read_three_byte_uint() {
         let data = &[0b0011_1100, 0b1000_0111, 0b1000_0001];
-        let uint = UInt::read(&mut Cursor::new(data), data.len()).expect(READ_ERROR_MESSAGE);
+        let uint = DecodedUInt::read(&mut Cursor::new(data), data.len()).expect(READ_ERROR_MESSAGE);
         assert_eq!(uint.size_in_bytes(), 3);
         assert_eq!(uint.value(), 3_966_849);
     }
```

Comment:
Not really related to your change, but I think it odd that I could have the following:

```rust
let data = &[0x00, 0x00, 0x00, 0x01];
let uint = DecodedUInt::read(...);
assert_eq(uint.size_in_bytes(), 4);
assert_eq(uint.value(), 1);
```

@zslayton, is this intentional?

Reply:
Yes. The `DecodedUInt` represents a value that was read from the stream; the `size_in_bytes` describes what was decoded. This API mirrors that of the `VarUInt` type, where the purpose for this is more obvious. 
```rust
let var_uint = VarUInt::read(some_input);
// How many bytes did we read from input?
let representation_size = var_uint.size_in_bytes();
// And what was the numeric value of the VarUInt?
let length = var_uint.value();
```
We don't _need_ this field/method per se in the `UInt` type because the reader had to specify how many bytes to read to begin with, it's mostly here for consistency with the other encoding primitives.